third_party/move/move-vm/runtime/src/data_cache.rs

```diff
@@ -27,19 +27,43 @@ use move_vm_types::{
 };
 use std::collections::btree_map::{BTreeMap, Entry};
 
+enum CachedInformation {
+    Value(GlobalValue),
+    SizeOnly(Option<u64>),
```

Comment:
nit: documentation to `SizeOnly` to explain why option?

Reply:
This will start to become CapturedReads :-) or we unite them before